### PR TITLE
Attestation demo

### PR DESCRIPTION
**Pull request by arp102**
This pull request adds a new custom demo client and a new method of connecting the client to the Pelz server. These additions are primarily based on the LocalAttestation sample code from the linux-sgx repo (https://github.com/intel/linux-sgx/tree/master/SampleCode/LocalAttestation), and much of the new code in this pull request is taken directly from that example with minimal changes.

The client and server establish an encrypted communication channel using an ECDH protocol defined in libsgx. During the key exchange, both parties are able to verify that they are running on the same SGX hardware instance, and they can also authenticate using additional metadata such as the MRSIGNER value.

Current status: The key exchange and encrypted communications are working properly (and have been tested in hardware mode), but the message handling is not fully operational yet, and there are several TODO comments that still need to be resolved. Primarily, the Pelz server code needs to be refactored to avoid nested ECALLs.

To run the demo client:
```bash
cd attestation_demo
make SGX_MODE=SIM
cd bin
./appinitiator
```

### CODE REVIEW

File: attestation_demo/EnclaveInitiator/Utility_E1.cpp [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+ */
+
+#include "sgx_eid.h"
+#include "EnclaveMessageExchange.h"
+#include "error_codes.h"
+#include "Utility_E1.h"
+#include "stdlib.h"
+#include "string.h"
+
+uint32_t marshal_input_parameters_e2_foo1(uint32_t target_fn_id, uint32_t msg_type, uint32_t var1, uint32_t var2, char** marshalled_buff, size_t* marshalled_buff_len)
```
Comment (b-carr) [inline attestation_demo/EnclaveInitiator/Utility_E1.cpp:39]: Are some of these autogenerated (I'm noticing "foo" in the function name, and variables "var1" and "var2". If so, what tooling did you use?

Comment (arp102) [reply attestation_demo/EnclaveInitiator/Utility_E1.cpp:39]: No, I don't think this was autogenerated, although I'm not sure why things were named that way.

**Comment by bit-bender** [comment]
So hopefully this achieved the intended goal - if not I will try to revert or correct these steps:

* I created PR #112 by cherry-picking the first commit (unmodified Intel local attestation demo code) and then merging it into the pelz main branch. This, however, resulted in a bunch of merge conflicts for this pull request.

* The 0dbf42f964d75b7caaaf7d7bdef9bdc0e5bb62cf commit represents the attempt to resolve the merge conflicts.

* When trying to build the code, I discovered that, in src/util/key_load.c, I must have inadvertently deleted the declaration for the 'common_name' charbuf and that the call to key_table_add_from_server was missing the 'client_name' parameter. The 55ad9c8926a74b5fa397a065425a0f5572f990d6 commit fixes these two specific issues.

File: attestation_demo/EnclaveInitiator/ReadMe.txt [hunk 2]
```diff
@@ -7,14 +7,14 @@ Enclave Project Wizard has created this Enclave1 project for you as a starting p
 This file contains a summary of what you will find in each of the files that make up your project.
```
Comment (b-carr) [inline attestation_demo/EnclaveInitiator/ReadMe.txt:7]: Do we need this file in the repo at all?

File: attestation_demo/README.txt [hunk 3]
```diff
@@ -1,5 +1,5 @@
 ---------------------------
```
Comment (b-carr) [inline attestation_demo/README.txt:1]: This file should be updated to reflect the files that actually get built, and how to use them with pelz.

**Review by b-carr — APPROVED** [review]
None of my comments are show-stoppers, so I'd be happy if we merged it as-is and took care of the documentation later (as long as we put in an issue to remind us.)

File: attestation_demo/EnclaveInitiator/ReadMe.txt (hunk 2 above)
Comment (arp102) [reply attestation_demo/EnclaveInitiator/ReadMe.txt:7]: No, I don't think so.

File: src/util/key_load.c [hunk 4]
```diff
@@ -216,6 +228,7 @@ int key_load(charbuf key_id)
   case PELZ_URI:
     {
       pelz_log(LOG_DEBUG, "Pelz Scheme Start");
+      charbuf common_name;
```
[the remark this answers is not in the document]
Comment (arp102) [reply src/util/key_load.c]: The changes with `common_name` were artifacts of the most recent merge with main.
It's fixed now.

File: attestation_demo/EnclaveInitiator/ReadMe.txt (hunk 2 above)
Comment (arp102) [reply attestation_demo/EnclaveInitiator/ReadMe.txt]: Removed in [bf683a5](https://github.com/NationalSecurityAgency/pelz/pull/107/commits/bf683a517abca63bb90cbdb4f5cd88478dbfc48f)

File: attestation_demo/README.txt (hunk 3 above)
Comment (arp102) [reply attestation_demo/README.txt]: Updated in [2b02c00](https://github.com/NationalSecurityAgency/pelz/pull/107/commits/2b02c00fb75bed6fea3a09063ae91ed448745314)

**Review by JWZeta — APPROVED** [review]

**Pull request merged**